components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -486,88 +486,363 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
     /**
      * Filter users using multi-attribute filters or single attribute filters with pagination.
      *
-     * @param node
-     * @param requiredAttributes
-     * @param offset
-     * @param limit
-     * @param sortBy
-     * @param sortOrder
-     * @param domainName
-     * @return
-     * @throws NotImplementedException
+     * @param node               Node
+     * @param requiredAttributes Required attributes
+     * @param offset             Starting index of the count
+     * @param limit              Number of required results (count)
+     * @param sortBy             Sortby
+     * @param sortOrder          Sort order
+     * @param domainName         Domain that the filter should perform
+     * @return Detailed user list
      * @throws CharonException
      */
     private List<Object> filterUsers(Node node, Map<String, Boolean> requiredAttributes, int offset, int limit,
-                                     String sortBy, String sortOrder, String domainName)
-            throws NotImplementedException, CharonException {
+            String sortBy, String sortOrder, String domainName) throws CharonException {
 
-        List<Object> filteredUsers = new ArrayList<>();
-        //0th index is to store total number of results
-        filteredUsers.add(0);
-        String[] userNames;
+        if (!(limit > 0))
+            limit = getMaxLimit();
```

Comment:
This limit was not enforced previously right. So it should have return all available. 
If so, we need to preserve that behavior